### PR TITLE
MAKAIRA-2201

### PR DESCRIPTION
**Pull request by duongnfq841992**

### CODE REVIEW

File: Client/Api.php [hunk 1]
```diff
@@ -150,4 +152,58 @@ static function ($hit) {
...
     }
+
+    /**
+     * @throws ConnectException
+     */
+    public function getRecommendedProducts(RecommendationQuery $query, string $debug = ''): array
+    {
+        $this->sanitizeLanguage($query);
+
+        $url = "$this->baseUrl/recommendation/public";
```
Comment (kyoya-de) [inline Client/Api.php]: For better readability: Please use curly braces, if you put a variable inside a string.

```suggestion
        $url = "{$this->baseUrl}/recommendation/public";
```

File: Client/Api.php [hunk 2]
```diff
@@ -150,4 +152,58 @@ static function ($hit) {
...
+
+        $headers = $this->defaultHeaders;
+        if (!empty($debug)) {
+            $headers[] = "X-Makaira-Trace: $debug";
```
Comment (kyoya-de) [inline Client/Api.php]: See above

```suggestion
            $headers[] = "X-Makaira-Trace: {$debug}";
```

File: Client/Api.php [hunk 3]
```diff
@@ -150,4 +152,58 @@ static function ($hit) {
...
+    /**
+     * @throws ConnectException
+     */
+    public function getMakairaRecommendations()
+    {
+        $url = "$this->baseUrl/reco";
+
+        $headers = $this->defaultHeaders;
+        if (!empty($debug)) {
+            $headers[] = "X-Makaira-Trace: $debug";
```
Comment (kyoya-de) [inline Client/Api.php]:
```suggestion
            $headers[] = "X-Makaira-Trace: {$debug}";
```

File: Client/Api.php [hunk 4]
```diff
@@ -150,4 +152,58 @@ static function ($hit) {
...
+
+        return $apiResult['items'];
+    }
+
+    /**
+     * @throws ConnectException
+     */
+    public function getMakairaRecommendations()
+    {
+        $url = "$this->baseUrl/reco";
```
Comment (kyoya-de) [inline Client/Api.php]:
```suggestion
        $url = "{$this->baseUrl}/reco";
```

File: Subscriber/JavascriptSubscriber.php [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'Theme_Compiler_Collect_Plugin_Javascript', 'collectJs'
+        ];
+    }
+
+    public function collectJs(\Enlight_Event_EventArgs $args)
+    {
+        $jsFiles = glob(__DIR__ . '../Resources/frontend/js/*.js');
```
Comment (kyoya-de) [inline Subscriber/JavascriptSubscriber.php]: `__DIR__` expands to the directory of the script without a tailing slash. It's possible that you forgot a slash `/`.

```suggestion
        $jsFiles = glob(__DIR__ . '/../Resources/frontend/js/*.js');
```

**Review by kyoya-de — CHANGES_REQUESTED** [review]

File: Subscriber/JavascriptSubscriber.php (hunk 5 above)
Comment (duongnfq841992) [reply Subscriber/JavascriptSubscriber.php]: You are right, the `__DIR__` constant doesn't have a trailing slash but my code is still working correctly. That's weird though.
Anyway, I fixed it.

File: Client/Api.php (hunk 4 above)
Comment (duongnfq841992) [reply Client/Api.php]: fixed

File: Client/Api.php (hunk 3 above)
Comment (duongnfq841992) [reply Client/Api.php]: fixed

File: Client/Api.php (hunk 2 above)
Comment (duongnfq841992) [reply Client/Api.php]: fixed

File: Client/Api.php (hunk 1 above)
Comment (duongnfq841992) [reply Client/Api.php]: fixed

**Review by kyoya-de — APPROVED** [review]

**Review by kyoya-de — APPROVED** [review]

**Pull request merged**